From 107f387ea625a67ef03b916ef965761f36de2bf4 Mon Sep 17 00:00:00 2001
From: Daniel Kahn Gillmor <dkg@fifthhorseman.net>
Date: Wed, 1 Jul 2020 17:15:12 -0400
Subject: [PATCH] Clarify warning message about PIC/PIE

As noted in discussion at https://dev.gnupg.org/T4982#135524, the
warning message produced when there is a mismatch between
position-independence of the Qt library and other compilations, the
warning produced by Qt is confusing.

This is an attempt to express a warning that is more closely aligned
with the actual test used.
---
 src/corelib/global/qglobal.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/corelib/global/qglobal.h b/src/corelib/global/qglobal.h
index fe8e8e8..971ee56 100644
--- a/src/corelib/global/qglobal.h
+++ b/src/corelib/global/qglobal.h
@@ -1280,7 +1280,7 @@ Q_CORE_EXPORT int qrand();
 #if !defined(QT_BOOTSTRAPPED) && defined(QT_REDUCE_RELOCATIONS) && defined(__ELF__) && \
     (!defined(__PIC__) || (defined(__PIE__) && defined(Q_CC_GNU) && Q_CC_GNU >= 500))
 #  error "You must build your code with position independent code if Qt was built with -reduce-relocations. "\
-         "Compile your code with -fPIC (-fPIE is not enough)."
+         "Compile your code with -fPIC (and not with -fPIE unless you have a very old version of GCC)."
 #endif
 
 namespace QtPrivate {
-- 
2.27.0

